<launch>

  <!-- GDB Debug Option -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg if="$(arg debug)" name="launch_prefix" value="gdb --args" />


  <!-- Arguments for ROS parameters -->
  <arg name="dataset" default="package://motion_bench_maker/problems/box_panda_flipped" />
  <arg name="results" default="package://motion_bench_maker/results/box_panda_flipped" />
  <arg name="exp_name" default="search_box_panda_flipped" />
  <arg name="train" default="False" />
  <arg name="time_limit" default="10" />

  <arg name="start" default="1" />
  <arg name="end" default="-1" />
  <arg name="runs" default="1" />

  <arg name="sensed" default="false" />
  <!-- <arg name="planners" default="wAstar_DEFAULT, wAstar_bfs_1_50, wAstar_je_1_50, ARAstar_DEFAULT, ARAstar_je_1_100_10_1, MHAstar_DEFAULT, MHAstar_je_bfs_1_50_2 "/> -->
  <arg name="planners" default="wPase_je_1_50"/>
  <!--   <arg name="planners" default="wAstar_DEFAULT"/> -->
  <!-- <arg name="planners" default="MHAstar_DEFAULT"/> -->
<!--   <arg name="planners" default="Epase_DEFAULT"/> -->


  <!-- Start the create dataset node -->
  <node  name="$(anon benchmark)" launch-prefix="$(arg launch_prefix)" pkg="motion_bench_maker" type="benchmark_search" output="screen" required = "true">

    <param name="dataset" value="$(arg dataset)" />
    <param name="exp_name" value="$(arg exp_name)" />
    <param name="sensed" value="$(arg sensed)" />
    <param name="results" value="$(arg results)" />
    <param name="start" value="$(arg start)" />
    <param name="end" value="$(arg end)" />
    <param name="runs" value="$(arg runs)" />
    <param name="time_limit" value="$(arg time_limit)" />
    <param name="train" value="$(arg train)" />
    <rosparam param="planners" subst_value="True">[$(arg planners)]</rosparam>
    
  </node>

</launch>
